sympy/core/power.py

```diff
@@ -440,6 +440,9 @@ def _eval_is_positive(self):
         elif self.base.is_nonpositive:
             if self.exp.is_odd:
                 return False
+            elif self.base.is_zero:
```

Comment:
Could this be indented as `elif self.base.is_nonpositive`?

Reply:
In that case, it must be before the `is_nonpositive` clause, right?

Reply:
since `is_zero` also implies `is_nonpositive` therefore, the indentation is correct. Also if `is_nonpositive` is handled then the control won't reach the `is_zero`.

Reply:
@oscargus is right. Plus I didn't want to have any big amount of changes in the core's code. I am still trying to understand it.

Reply:
This change seems safe to implement in core.

Reply:
> In that case, it must be before the is_nonpositive clause, right?

Yes, that is how I thought it should be done. It is a test on `base`, so its place could  be among the other `base` tests.

Reply:
I didn't find such test segment in `test_assumptions` file.

Reply:
> It is a test on `base`, so its place could be among the other `base` tests.

Makes sense.

@ShubhamKJha I understand your hesitation to do larger changes in core. But it would make sense to test `is_zero` before `is_nonpositive` for consistency reasons.

Reply:
@oscargus If you suggest I will change it. 

> It is a test on `base`, so its place could be among the other `base` tests.

I can't even find tests that check if a `Pow` object is positive or negative. I think I will have to add new tests for them.

Reply:
I was referring to the ìf`-clause.

```
elif self.base.is_zero:
    Your new code
elif self.base.is_nonpositive:
   The old code
elif ...
```

(Similar for nonnegative.)

(And as there will be a change, I just note that the brackets around `(self.exp.is_zero is False)`, as you noticed later.)

Reply:
The brackets are unwanted since the precedence of `is` is more than `and`. I will remove them and modify the code as you suggested. I kept it there as I thought this is one of the special cases and shouldn't be called always. Instead `self.base.is_nonpositive` would be mostly required. But since the overhead of this change is negligible, I can change it.

Reply:
How does it handles the case when base is non-positive and exponent is even?

Reply:
It doesn't matter if the exponent is even. All real numbers should produce the same result.

Reply:
Okay, can you just clear in which if-elif branch does `(S(-2)**S(2)).is_positive` will fall?

Reply:
Oh I got it. :-)

Reply:
I think you mistyped.